### PR TITLE
Drop old sf and add type hints

### PR DESCRIPTION
**Pull request by greg0ire**
This PR removes support for old versions of Symfony, but there might still be code that can be removed. Also, type hints were added where obvious. To be merged before #113

### CODE REVIEW

**Comment by greg0ire** [comment]
Ah I should have run the tests locally, there are a lot of seemingly hard to fix issues.

File: PhpUnit/ContainerBuilderHasServiceDefinitionConstraint.php [hunk 1]
```diff
@@ -12,16 +12,11 @@ class ContainerBuilderHasServiceDefinitionConstraint extends Constraint
...
-        }
-
-        if ($checkExpectedClass && !is_string($expectedClass)) {
-            throw new \InvalidArgumentException('The $expectedClass argument should be a string');
-        }
-
+    public function __construct(
+        string $serviceId,
+        ?string $expectedClass = null,
+        $checkExpectedClass = true
```
Comment (Nyholm) [inline PhpUnit/ContainerBuilderHasServiceDefinitionConstraint.php]: bool ?

Comment (greg0ire) [reply PhpUnit/ContainerBuilderHasServiceDefinitionConstraint.php]: Ok, I will add `bool` here, and also dive into the rabbit hole of adding less obvious type hints.

Comment (greg0ire) [reply PhpUnit/ContainerBuilderHasServiceDefinitionConstraint.php]: Ḩ̶͚̗͙̻̝͖̠̹̊́̂̓̆̀͑͋́͆͊̚͝ë̷̦̪́́͂̎̓̇̈́̃̐̑͝ṟ̵̨̲͚͚̻͎̫͉̊é̴̹͎̺̺͔̤̇̅͐̂͗̊͠͠͝ ̴̨̨̬̗̥̥̟̙̣͚͌̀͂̎̃͋̆̿̎͒́̄̾̐͝ì̸͚͚̲̜͈͈̓t̶͔̿͆̅̇͌̒̑̎͊ ̴͖̹͚̮̍̓̈̓́ͅḉ̶̛̩̙̙̽̐̇̇͝͝ͅo̷̡͎̙͑͆̆͊̋̎̋͋̚͝͝͝m̷̧̨̖̱͔̺̼͎̘̒͗̾̆̔̀́̿̇̚ȇ̶͎͝s̸̢̤͓͚͔̹̏́̈́̏͆͂!̵͓̯͚̮̩̻̯̱̻̥̝̭͊̐͐̾̚͠ ̴̘̻̼̰̫͙̇͌͂̋͆͜͝T̸̢̲͔̼̹͛̃̉̉̓̉̀̔͐̎̑͊̕͘͝h̸̤̭̱͎̼̺̟͇̭͚̤͌͌͛̿̎̚ͅͅè̵̛̳͉̖̤̥̬ ̵̢̧̗̝̻̥̲͙̱̫͓̣̪̮̂̽̈́͜h̵̲̭̱͖͊͂̓̓́͘u̵̹̔̀́͐̍̈͘̕g̷͔̳͇͙͔̩̩̙̊e̷͎͚̱͇͆̔̉̎͂͆̇͆̉͋̓͗̔͝ ̴̧͙̪̙̯̏͂́̈p̶̧̘̯͖̼̖̼̓̇͐̉͐̃̊̅̓̅̏̕̚u̸͓̓̋̿̿̑͝ļ̸̰͖̮̬͔̝̳͉̘͔̍̈́́̐̄̈̋̀̌̔̈́͘͝l̴̯̭͌̆́̊̓̐̓ ̸̧̨̹̝̠̤̜̜̘̑̓͂r̸̢͍̘͖̗̻͋̓͛̓̃̇̓̄͋͗́͘͘͠e̵͓̠̟̬̞̎́͊̂̾͘q̵̛͓̊̇̊͆̊̎͗̓͜ṳ̸̹̼͉̭͇̬̹͇͉̏̇̆̿̀̈́̏͐̈͘ḙ̸̢̢̨̡̛̜̖̪̤̤̫̣͉͍̅̎̀̔͒̎̊̽͆͐̕͠ͅs̴̨̡̛̬͓̘̰̗̗̒̀́t̷̡̡̡͖͕͙̥͈̩̲̱̳̪̉̑̃͆̅́͒̿́̈́̚͝!̷̤̙̣̣͎̗̤̟͕̅͆̀̓͛̊̈ͅͅ ̷̬̒͐̏̍̃͆͂̀̓̋̑̕͜B̷̧͇̑̆̀̀̎̈́̋̃͑͐͋͊̀̕e̸͈͈͙̼͕̳̞͈͈͉̙͋͂̽̄̉͛̉̓̇̐͑͘͠h̶̨̢̡̙̜͙͍͈̫̹͌̂ô̸̤͊l̴̢̡̨͓͖̱̪̝̪̗͗̇͒̈̐̾̈́̍̅͆͠͝d̸̢̬̺̻̜̣̉͛̓!̵̢̢̳͉̱͉͉̰̗̪͍̜̺̲̈́̓̾͐̐̉̄̑̈́

**Comment by Nyholm** [comment]
Im happy to merge this PR if you are done

**Comment by greg0ire** [comment]
I am, please proceed :)

**Pull request merged**